src/main/java/elysia/ui/FileReaderWriter.java

```diff
@@ -0,0 +1,83 @@
+package elysia.ui;
+
+import elysia.tasks.Deadline;
+import elysia.tasks.Event;
+import elysia.tasks.TaskList;
+import elysia.tasks.Todo;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class FileReaderWriter {
+    java.nio.file.Path path;
+    TaskList taskList;
+
+    public FileReaderWriter(TaskList taskList) {
+        path = java.nio.file.Paths.get("data.txt");
+        this.taskList = taskList;
+    }
+
+    public String createFile() {
+        String output = "";
+        try {
+            File myObj = new File(String.valueOf(path));
+            if (myObj.createNewFile()) {
+                output = ("I created " + myObj.getName() + " for you~");
+            }
+        } catch (IOException e) {
+            output = "Aww something went wrong :(\n";
+        }
+        return output;
+    }
+
+    public String writeFile() {
+        String output = "";
+        try {
+            Files.writeString(path, taskList.toFile(), StandardCharsets.UTF_8);
+            output = "Saving all your tasks for you~";
+        } catch (IOException e) {
+            output = "Aww something went wrong :(\n";
+        }
+        return output;
+    }
+
+    public String readFile() {
+        String output = "";
+        File file = new File(path.toString());
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNextLine()) {
+                String taskInput = sc.nextLine();
+                String[] taskParts = taskInput.split("\\|");
+                switch(taskParts[0]) {
+                case "T":
```

Comment:
Great to see that you have indented your case statements inside the switch block.